packages/gravity/blade-runner/src/replicants/echo-replicant.ts

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { Trigger } from '@dxos/async';
+import { type AutomergeUrl } from '@dxos/automerge/automerge-repo';
+import { Filter, type QueryResult, type EchoDatabaseImpl } from '@dxos/echo-db';
+import { EchoTestPeer } from '@dxos/echo-db/testing';
+import { type ReactiveObject, S, TypedObject, create } from '@dxos/echo-schema';
+import { invariant } from '@dxos/invariant';
+import { PublicKey } from '@dxos/keys';
+import { createTestLevel } from '@dxos/kv-store/testing';
+import { log } from '@dxos/log';
+
+import { ReplicantRegistry } from '../plan';
+
+export class Text extends TypedObject({ typename: 'dxos.blade-runner.Text', version: '0.1.0' })({
+  content: S.string,
+}) {}
+
+export class EchoReplicant {
+  private _testPeer?: EchoTestPeer = undefined;
+  private _db?: EchoDatabaseImpl = undefined;
+
+  async open({
+    spaceKey = PublicKey.random().toHex(),
+    path,
+    rootUrl,
+  }: { spaceKey?: string; path?: string; rootUrl?: AutomergeUrl } = {}) {
+    this._testPeer = new EchoTestPeer(path ? createTestLevel(path) : undefined);
+    await this._testPeer.open();
+    this._db = rootUrl
+      ? await this._testPeer.openDatabase(PublicKey.fromHex(spaceKey), rootUrl)
+      : await this._testPeer.createDatabase(PublicKey.fromHex(spaceKey));
+    this._db.graph.runtimeSchemaRegistry.registerSchema(Text);
+
+    log.trace('dxos.echo-replicant.open', { spaceKey, path });
+    return {
+      spaceKey: this._db.spaceKey.toHex(),
+      rootUrl: this._db.rootUrl,
+    };
+  }
+
+  async close(): Promise<void> {
+    await this._db!.close();
+    await this._testPeer!.close();
+  }
+
+  async createDocuments({
```

Comment:
is this called in this PR?
i don't understand the replicant API or where the spec is defined that is used to call these methods. can you DM me an example?

Reply:
see https://github.com/dxos/dxos/blob/mykola-dm/refactor-blade-runner/packages/gravity/blade-runner/src/spec/echo.ts
```
    const replicant = await env.spawn(EchoReplicant, { platform: 'nodejs' });
    await replicant.brain.createDocuments({
        amount: params.spec.numberOfObjects,
        insertions: params.spec.numberOfInsertions,
        mutationsSize: params.spec.insertionSize,
      });
```